deep-learning/_posts/2018-04-12-spars_ensemble.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: review
+title:  "LEARNING SPARSE STRUCTURED ENSEMBLES"
+tags:   "Deep Learning"
+author: Faezeh Amjadi
+pdf:    https://arxiv.org/abs/1803.00184
+cite:
+authors: "Yichi Zhang, Zhijian Ou"
+title:   " Learning Sparse Structured Ensembles with SG-MCMC and Network Pruning"
+venue:   "arXiv"
+---
+
+### Description
+
+In this work, learning a Sparse Structure Ensemble within a single training process by Bayesian posterior sampling on model parameters is proposed in two phase to acheive a high prediction accuracy and reduce memory and computation cost. 
+The authors, in the first phase, run an SG-MCMC to draw an ensemble of samples from the posterior distribution
+of network parameters. In the second phase, weight-pruning is applied to each sampled networks and then retraining over the remained connections.
+For each test point $hat{x}$, predictive distribution is:
+
+<img src="/deep-learning/images/mcmc/1.png" width="400">
+
+<img src="/deep-learning/images/mcmc/7.png" width="800">
+
+
+The parameters are approximated by Mont Carlo which will be sampled from posterior distribution of network parameters.
+
+
+###  (SG-MCMC) sampling:
+MCMC is a common sampling but is not good for large datasets. Stochastic Gradient Markov Chain Monte Carlo (SG-MCMC) represents a family of MCMC sampling algorithms which handling large datasets efficiently and works by adding a scaled gradient noise based on minibatchs and thus improve exploration of the model parameter space. $U$ is known as the potential energy and $hat{U}$ is its approximation over  t-th mini bach and \(\epsilon\) is learning rate.
+
+
+
+<img src="/deep-learning/images/mcmc/2.png" width="600">
+
+
+
+After all the models are collected, authors use a simple pruning rule to reduce the model size as well as the computation cost with finding the network connections whose weights are below a certain threshold and removing them away. Now coming to next phase, with multiple sparse models, there is zero positions in weight matrics is an issue, so authors use Sparse Structure Ensemble (SSE) which employe group lasso regularization, proposes to do feature selection in group level, which means keeping or removing all the parameters in a group simultaneously to achieve structured sparsity corresponding to grouping strategy and penalize unimportant filter and channels. So, group lasso regularization can be formulated as :
```

Comment:
to the 2nd phase (model combination) ...

I do not understand "there is zero positions in weight matrics is an issue"  Please reformulate

Reply:
employs

Reply:
Please shorten the size of your sentences.  One sentence shouldn't be more than 1 or 2 lines.